compiler/src/dmd/expressionsem.d

```diff
@@ -1155,6 +1155,43 @@ L1:
     return e1;
 }
 
+/*
+ * Check whether `outerFunc` and `calledFunc` have the same `this`.
+ * If `calledFunc` is the member of a base class of the class that contains
+ * `outerFunc` we consider that they have the same this.
+ *
+ * Returns:
+ *  `true` if outerFunc and calledFunc may use the same `this` pointer.
+ * `false` otherwise.
+ */
+private bool haveSameThis(FuncDeclaration outerFunc, FuncDeclaration calledFunc)
+{
+    auto thisAd = outerFunc.isMemberLocal();
+    if (!thisAd)
+        return false;
+
+    auto requiredAd = calledFunc.isMemberLocal();
+    if (!requiredAd)
+        return false;
+
+    if (thisAd == requiredAd)
```

Comment:
wouldn't just `is` be enough here ?

Reply:
Indeed, `is` would be preferred since we want to compare the pointers, but AST nodes do not define opEquals so we are fine with the default opEquals of object.

Long story-short: both is and == are fine in this case.

Reply:
I dont see the problem with `is`, for the comment about coverage you are protective, i.e "in case of" for but this one you are not.